app/models/hyrax/file_metadata.rb

```diff
@@ -1,6 +1,25 @@
 # frozen_string_literal: true
 
 module Hyrax
+  ##
+  # Casts a resource to an associated FileMetadata
+  #
+  # @param [Valkyrie::StorageAdapter::File] file
+  #
+  # @return [Hyrax::FileMetadata]
+  # @raise [ArgumentError]
+  def self.FileMetadata(file)
+    raise(ArgumentError, "Expected a Valkyrie File; got #{file.class}: #{file}") if
```

Comment:
`Expected a Valkyrie::StorageAdapter::File` might be even clearer

Reply:
```suggestion
    raise(ArgumentError, "Expected a Valkyrie::StorageAdapter::File; got #{file.class}: #{file}") if
```